src/main/java/org/spongepowered/api/event/world/chunk/WorldChunkEvent.java

```diff
@@ -86,14 +86,14 @@
     /**
      * Called when a new {@link WorldChunk} was generated.
      */
-    interface Generated extends ChunkEvent {
+    interface Generated extends WorldChunkEvent {
```

Comment:
Is this actually being called right now? I have the feeling that Generated is somewhat stepping onto the point of being loaded since a chunk isn't "generated" until it flows through the pipeline, but then it's not a WorldChunk yet.

Reply:
https://github.com/SpongePowered/Sponge/blob/e84a9c53f48f2a3a9c2b485b5ed7176e9ed6020b/src/mixins/java/org/spongepowered/common/mixin/core/server/level/ChunkMapMixin.java#L137-L149

It's done so here and uses an `ImposterProtoChunk` which wraps a `LevelChunk`. I haven't personally looked at this code though.